content/articles/2019-05-itdepends.Rmarkdown

```diff
@@ -0,0 +1,241 @@
+dependencies as our metric.
+
+## prefer a balanced weight
+
+Instead of striving for absolutely minimal dependencies I would instead suggest
+a more holistic, balanced and quantitative approach.
+
+A holistic approach looks at the project as a whole and asks who is the primary
+audience? If the audience is other package authors than a leaner package with fewer
+dependencies may be more appropriate. If instead it is primarily data scientists or
+statisticians they will likely already have many popular dependencies installed and
+a more feature-full package is preferred.
+
+A balanced approach understands that adding (or removing) dependencies
+come with trade-offs. Adding a dependency gives you additional features,
+bug fixes, and real-world testing, at the costs of increased installation time,
+disk space and maintenance if the dependency has breaking changes.
+
+In contrast, removing a dependency lowers installation time, disk space and
+avoids potential breaking changes. But at the cost of fewer features, an
+increase in bugs and _additional_ development time.
+
+A quantitative approach lets you more accurately measure your dependencies, and
+see which are heavier or lighter along various axes. Measuring each of the
+dependencies gives you more power to reason about which dependencies would be
+most beneficial to remove (if any).
+
+How you weigh these costs and benefits varies by the skills and experience of
+the maintainer and should also vary by the goals of the project.
+
+## beware illusionary superiority
+
+When making these decisions it is important to consider the
+psychological phenomenon of [Illusionary
+Superiority](https://en.wikipedia.org/wiki/Illusory_superiority). Illusory
+superiority is the idea that people (regardless of background) overestimate
+(sometimes greatly) their own abilities in a given field. No one is truly
+immune to this phenomenon, including the author of this post! This phenomenon
+will cause you to overestimate your ability to re-implement features and
+underestimate the introduction of new bugs.
+
+However even if you truly _are_ a more skilled programmer than the maintainer of
+a dependency I would still advise caution. The existing package has the
+significant benefits of _time_ and _users_. Users who have undoubtedly uncovered
+numerous bugs using the package in the real world, which are now fixed.
+
+## itdepends, a package to help with dependencies
+
+The [itdepends](https://github.com/jimhester/itdepends) package provides tools
+to assess usage, measure weights, visualize proportions and if needed, assist
+removal of dependencies.
+
+## Assess usage with `itdepends::dep_usage()`
+
+`dep_usage_proj()` and `dep_usage_pkg()` return a tibble of
+every function call in your project or package. This information can then be
+used to do things like
+
+- show which packages you are using the most in a project
+
+```{r}
+library(tidyverse)
+itdepends::dep_usage("~/p/tidyversedashboard") %>%
```

Comment:
Might be a worth a `create_from_github()` to make more reproducible?

Reply:
I switched this to use `dep_usage_pkg()` so it is reproducible across environments and doesn't depend on a specific project.